src/adapter/src/catalog/storage.rs

```diff
@@ -2037,15 +2036,110 @@ pub struct ClusterReplica {
+// separate structs. As of writing this comment, it is mainly due to the fact that we don't persist
+// the replica allocation.
+
+#[derive(Clone, Debug, PartialOrd, PartialEq, Eq, Ord)]
+pub struct ReplicaConfig {
+    pub location: ReplicaLocation,
+    pub logging: ReplicaLogging,
+    pub idle_arrangement_merge_effort: Option<u32>,
+}
+
+impl From<mz_controller::clusters::ReplicaConfig> for ReplicaConfig {
+    fn from(config: mz_controller::clusters::ReplicaConfig) -> Self {
+        Self {
+            location: config.location.into(),
+            logging: config.compute.logging,
+            idle_arrangement_merge_effort: config.compute.idle_arrangement_merge_effort,
+        }
+    }
+}
+
+#[derive(Debug, Clone, PartialOrd, PartialEq, Eq, Ord)]
+pub enum ReplicaLocation {
+    Unmanaged {
+        storagectl_addrs: Vec<String>,
+        storage_addrs: Vec<String>,
+        computectl_addrs: Vec<String>,
+        compute_addrs: Vec<String>,
+        workers: usize,
+    },
+    Managed {
+        size: String,
+        /// `Some(az)` if the AZ was specified by the user and must be respected;
+        availability_zone: Option<String>,
+        disk: bool,
+    },
+}
+
+impl From<mz_controller::clusters::ReplicaLocation> for ReplicaLocation {
+    fn from(loc: mz_controller::clusters::ReplicaLocation) -> Self {
+        match loc {
+            mz_controller::clusters::ReplicaLocation::Unmanaged(
+                mz_controller::clusters::UnmanagedReplicaLocation {
+                    storagectl_addrs,
+                    storage_addrs,
+                    computectl_addrs,
+                    compute_addrs,
+                    workers,
+                },
+            ) => Self::Unmanaged {
+                storagectl_addrs,
+                storage_addrs,
+                computectl_addrs,
+                compute_addrs,
+                workers,
+            },
+            mz_controller::clusters::ReplicaLocation::Managed(
+                mz_controller::clusters::ManagedReplicaLocation {
+                    allocation: _,
+                    size,
+                    availability_zones,
+                    disk,
+                },
+            ) => ReplicaLocation::Managed {
+                size,
+                availability_zone:
+                    if let mz_controller::clusters::ManagedReplicaAvailabilityZones::FromReplica(
+                        Some(az),
+                    ) = availability_zones
+                    {
+                        Some(az)
+                    } else {
+                        None
+                    },
+                disk,
+            },
+        }
+    }
+}
+
+#[derive(Clone, Debug)]
+pub struct ComputeReplicaLogging {
+    pub log_logging: bool,
+    pub interval: Option<Duration>,
+}
+
+impl From<mz_compute_client::controller::ComputeReplicaLogging> for ComputeReplicaLogging {
+    fn from(logging: mz_compute_client::controller::ComputeReplicaLogging) -> Self {
```

Comment:
This code seems untested, https://buildkite.com/materialize/coverage/builds/216#_

Reply:
Thanks, that was dead code. I've removed it.